hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java

```diff
@@ -1847,10 +1848,12 @@ public MetadataBuildingContext getBuildingContext() {
 								}
 							}
 					);
-
-					fk.setName( nameIdentifier.render( getDatabase().getJdbcEnvironment().getDialect() ) );
+				} else {
+					nameIdentifier = Identifier.toIdentifier( fk.getName(), true );
 				}
 
+				fk.setName( getMetadataBuildingOptions().getPhysicalNamingStrategy().toPhysicalSequenceName(nameIdentifier, getDatabase().getJdbcEnvironment()).render( getDatabase().getJdbcEnvironment().getDialect() ) );
```

Comment:
Totally unexpected to route this through `#toPhysicalSequenceName`.  A FK is not a sequence ;)